plugins/zstack.py

```diff
@@ -332,14 +342,21 @@ def initAcquisition(self):
         self.zstart.unsubscribe(self._on_zstart)
         return self._estimate_step_duration()
 
-    def stepAcquisition(self, i, images):
+    def preStepAcquisition(self, i):
         """
-        An action that executes for the ith step of the acquisition
-        i (int): the step number
+        Called before the ith step of the acquisition
+        i (0 <= int): the step number
+        """
+        self.focus.moveAbs({'z': self.zstart.value + self.zstep.value * i}).result()
+
+    def postStepAcquisition(self, i, images):
+        """
+        Called after the ith step of the acquisition
+        i (0 <= int): the step number
         images []: A list of images as DataArrays
         """
-        self.focus.moveRel({'z': self.zstep.value}).result()
-        
+        pass
```

Comment:
Can you add a comment why you're  not doing anything here?

Reply:
Yes, indeed it's a little odd. I've added a comment.
The main reason that this function is there is that the plugin has been built in a way that it should be generic to handle any axis move, by just inheriting the class. It hasn't been used yet, but the function is there in case another type of acquisition would need it....